src/main/core/worker.rs

```diff
@@ -680,6 +680,26 @@ mod export {
         Worker::is_alive()
     }
 
+    #[no_mangle]
+    pub extern "C" fn worker_resolveIPToAddress(ip: libc::in_addr_t) -> *const cshadow::Address {
+        Worker::with_mut(|w| {
```

Comment:
Optional - maybe have these use `with_ref` instead of `with_mut`? We'd either need to cast the dns  pointer to a mut pointer here, with an explanation that we know the C function doesn't actually mutate, or move the cast inside the C function when calling the hash table API.

Reply:
Good catch, fixed (changed to `with` instead of `with_mut`)!